### DISCUSSION TITLE
Complex rule for single fact

### DISCUSSION THREAD

**Discussion by suresh-webonise** [CacheControl/json-rules-engine#327] [General]
Suppose the fact is
const fact={
  id:1,
  username:'sureshk',
  student:[{name:'zbc',percentage:100},{name:'zb',percentage:10}],
  role:['coach','mentor'],
  sport:['hockey','base']
}
and the rule should be 

rule criteria => (student.name ='zbc' AND student.percentage=100) AND id=1 AND username='sureshk' AND role = 'coach' AND sport='hockey'
what will the json rules for the above rule criteria and facts

**Comment by chris-pardy** [comment]
@suresh-webonise it's not currently possible to express this in json-rules-engine without adding a custom operator. Specifically the part where you want to assert the one of the entries in `student` has both a specific name and specific precent is not possible with the included operators.
